File: backend/Bingol-University-Social-Media-main/src/main/java/bingol/campus/chat/dto/PrivateChatDTO.java
package bingol.campus.chat.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;


@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class PrivateChatDTO {
    private UUID chatId;

    private String chatName;
    private String chatPhoto;

    private String username1;

    private String username2;

    private MessageDTO lastEndMessage;
}

File: frontend/lib/models/private_chat_dto.dart
import 'package:json_annotation/json_annotation.dart';
import 'message_dto.dart'; // MessageDTO'yu içe aktarın

part 'private_chat_dto.g.dart';

@JsonSerializable()
class PrivateChatDTO {
  final String chatId; // UUID'ler String olarak temsil edilir
  final String chatName; // Sohbet adı
  final String chatPhoto; // Sohbet fotoğrafı
  final String username1; // Kullanıcı 1
  final String username2; // Kullanıcı 2
  final MessageDTO lastEndMessage; // Son mesaj

  PrivateChatDTO({
    required this.chatId,
    required this.chatName,
    required this.chatPhoto,
    required this.username1,
    required this.username2,
    required this.lastEndMessage,
  });

  factory PrivateChatDTO.fromJson(Map<String, dynamic> json) => _$PrivateChatDTOFromJson(json);
  Map<String, dynamic> toJson() => _$PrivateChatDTOToJson(this);
} 